model/diagnostic/DiagnosticDataTable.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ *
+ */
+
+namespace oat\taoClientDiagnostic\model\diagnostic;
+
+use DateTime;
+use oat\taoClientDiagnostic\exception\StorageException;
+use oat\taoClientDiagnostic\model\storage\PaginatedStorage;
+use tao_helpers_Date as DateHelper;
+use oat\taoClientDiagnostic\model\storage\Storage;
+use oat\taoClientDiagnostic\model\storage\PaginatedSqlStorage;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+/**
+ * This temporary helpers is a temporary way to return data to the controller.
+ * This helps isolating the mock code from the real controller one.
+ * It will be replaced by a real service afterward.
+ */
+class DiagnosticDataTable implements ServiceLocatorAwareInterface
```

Comment:
Already discussed with JS, he will refactor this part to improve that and able to work with paginated storage